Panic while parsing message params

Copied from slack (https://filecoinproject.slack.com/archives/C0174P5M11T/p1646375434677989)

Hi there! Encountered an error with lily (calibrationnet) and walk job. When the job tries to process block 435326  lily daemon restarts with the following in the logs:

```
panic: reflect: call of reflect.Value.NumField on zero Value [recovered]
        panic: reflect: call of reflect.Value.NumField on zero Value [recovered]
        panic: reflect: call of reflect.Value.NumField on zero Value

goroutine 1193406 [running]:
encoding/json.(*encodeState).marshal.func1()
        /usr/local/go/src/encoding/json/encode.go:328 +0x6e
panic({0x3269820, 0xc03d0d0378})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
encoding/json.(*encodeState).marshal.func1()
        /usr/local/go/src/encoding/json/encode.go:328 +0x6e
panic({0x3269820, 0xc03d0d0378})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
reflect.flag.mustBe(...)
        /usr/local/go/src/reflect/value.go:223
reflect.Value.NumField({0x0, 0x0, 0x32d7920})
        /usr/local/go/src/reflect/value.go:1723 +0x8d
github.com/filecoin-project/lily/lens/util.(*paramWrapperType).MarshalJSON(0xc03d0d0258)
        /go/src/github.com/filecoin-project/lily/lens/util/repo.go:393 +0x3e7
encoding/json.marshalerEncoder(0xc032a5a580, {0x32556c0, 0xc03d0d0258, 0x618ea9}, {0xb, 0x0})
        /usr/local/go/src/encoding/json/encode.go:479 +0xbe
encoding/json.(*encodeState).reflectValue(0x324a1c0, {0x32556c0, 0xc03d0d0258, 0xc046e910e0}, {0x0, 0x3})
        /usr/local/go/src/encoding/json/encode.go:360 +0x78
encoding/json.interfaceEncoder(0xc032a5a580, {0x324a1c0, 0xc03ea95970, 0x1}, {0xe9, 0xd1})
        /usr/local/go/src/encoding/json/encode.go:716 +0xc8
encoding/json.mapEncoder.encode({0xc00007e800}, 0xc032a5a580, {0x32933c0, 0xc046e910b0, 0x32933c0}, {0xc7, 0x7c})
        /usr/local/go/src/encoding/json/encode.go:814 +0x5c3
encoding/json.(*encodeState).reflectValue(0xc0266bc8e8, {0x32933c0, 0xc046e910b0, 0x0}, {0xc0, 0xe9})
        /usr/local/go/src/encoding/json/encode.go:360 +0x78
encoding/json.(*encodeState).marshal(0x18, {0x32933c0, 0xc046e910b0}, {0x0, 0x0})
        /usr/local/go/src/encoding/json/encode.go:332 +0xfa
encoding/json.Marshal({0x32933c0, 0xc046e910b0})
        /usr/local/go/src/encoding/json/encode.go:161 +0x45
github.com/filecoin-project/lily/lens/util.(*paramWrapperType).MarshalJSON(0xc03d0d0108)
        /go/src/github.com/filecoin-project/lily/lens/util/repo.go:401 +0x5e9
encoding/json.marshalerEncoder(0xc049e29280, {0x32556c0, 0xc03d0d0108, 0x618ea9}, {0x2, 0x0})
        /usr/local/go/src/encoding/json/encode.go:479 +0xbe
encoding/json.(*encodeState).reflectValue(0x324a1c0, {0x32556c0, 0xc03d0d0108, 0xc046e91050}, {0x80, 0x1})
        /usr/local/go/src/encoding/json/encode.go:360 +0x78
encoding/json.interfaceEncoder(0xc049e29280, {0x324a1c0, 0xc03ea958c0, 0x569001}, {0x68, 0x1a})
        /usr/local/go/src/encoding/json/encode.go:716 +0xc8
encoding/json.mapEncoder.encode({0xc045ccb3b0}, 0xc049e29280, {0x32933c0, 0xc046e91020, 0x32933c0}, {0x0, 0xcf})
        /usr/local/go/src/encoding/json/encode.go:814 +0x5c3
encoding/json.(*encodeState).reflectValue(0xc0266bcf70, {0x32933c0, 0xc046e91020, 0x7}, {0xc0, 0xe9})
        /usr/local/go/src/encoding/json/encode.go:360 +0x78
encoding/json.(*encodeState).marshal(0x18, {0x32933c0, 0xc046e91020}, {0x0, 0x0})
        /usr/local/go/src/encoding/json/encode.go:332 +0xfa
encoding/json.Marshal({0x32933c0, 0xc046e91020})
        /usr/local/go/src/encoding/json/encode.go:161 +0x45
github.com/filecoin-project/lily/lens/util.(*paramWrapperType).MarshalJSON(0xc0266bd318)
        /go/src/github.com/filecoin-project/lily/lens/util/repo.go:401 +0x5e9
github.com/filecoin-project/lily/lens/util.MarshalWithOverrides(...)
        /go/src/github.com/filecoin-project/lily/lens/util/repo.go:359
github.com/filecoin-project/lily/lens/util.ParseParams({0xc02fc401b0, 0x8c, 0x8c}, 0xc0266bd4d8, {{0xc03bb9ea20, 0xc0266bd4a8}})
        /go/src/github.com/filecoin-project/lily/lens/util/repo.go:241 +0x4f4
github.com/filecoin-project/lily/lens/util.MethodAndParamsForMessage(0xc0384a1dc0, {{0xc03bb9ea20, 0xb}})
        /go/src/github.com/filecoin-project/lily/lens/util/repo.go:258 +0x7b
github.com/filecoin-project/lily/tasks/messages.(*Task).parseMessageParams(...)
        /go/src/github.com/filecoin-project/lily/tasks/messages/message.go:289
github.com/filecoin-project/lily/tasks/messages.(*Task).ProcessMessages(0x1, {0x4126de0, 0xc046e902d0}, 0xc043404d80, 0xc0434051c0, {0xc001d92000, 0x27, 0xc000234b90}, {0xc03ff101c0, 0x2, ...})
        /go/src/github.com/filecoin-project/lily/tasks/messages/message.go:218 +0x1aac
github.com/filecoin-project/lily/chain.(*TipSetIndexer).runMessageProcessor(0x0, {0x4126da8, 0xc04bbe1020}, {0x40d9540, 0x5b3adc0}, {0x364da48, 0x40d94c0}, 0xc043404d80, 0x3647b87, {0xc001d92000, ...}, ...)
        /go/src/github.com/filecoin-project/lily/chain/indexer.go:677 +0x378
created by github.com/filecoin-project/lily/chain.(*TipSetIndexer).TipSet
        /go/src/github.com/filecoin-project/lily/chain/indexer.go:258 +0x4308
```

If I skip that particular block, everything goes well

Job tasks:
```
--tasks=blocks,messages,chaineconomics,msapprovals,actorstatesraw,actorstatesmultisig
```
Storage: TimescaleDB

--- a/lens/util/repo.go
+++ b/lens/util/repo.go
@@ -241,6 +241,9 @@ func ParseParams(params []byte, method abi.MethodNum, actCode cid.Cid) (string, 
 	b, err := MarshalWithOverrides(p, map[reflect.Type]marshaller{
 		reflect.TypeOf(bitfield.BitField{}): bitfieldCountMarshaller,
 	})
+	if err != nil {
+		return "", "", xerrors.Errorf("failed to parse message params method: %d, actor code: %s, params: %s: %w", method, actCode, string(params), err)
+	}
 
 	return string(b), m.Name, err
 }
@@ -351,7 +354,13 @@ func MakeGetActorCodeFunc(ctx context.Context, store adt.Store, next, current *t
 
 type marshaller func(interface{}) ([]byte, error)
 
-func MarshalWithOverrides(v interface{}, overrides map[reflect.Type]marshaller) ([]byte, error) {
+func MarshalWithOverrides(v interface{}, overrides map[reflect.Type]marshaller) (out []byte, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			out = nil
+			err = xerrors.Errorf("failed to override message param json marshaller: %v", r)
+		}
+	}()
 	pwt := paramWrapperType{
 		obj:     v,
 		replace: overrides,
@@ -388,6 +397,10 @@ func (wt *paramWrapperType) MarshalJSON() ([]byte, error) {
 
 	switch t.Kind() {
 	case reflect.Struct:
+		// if an empty struct marshal the empty struct and bail.
+		if v.IsZero() {
+			return json.Marshal(wt.obj)
+		}
 		// if its a struct, walk its fields and recurse.
 		m := make(map[string]interface{})
 		for i := 0; i < v.NumField(); i++ {

